File: ExcelWordVB/Form1.vb
Imports System.IO
Imports Microsoft.Office.Interop
''Crear una aplicacion en VS  (vb y c#) para generar un documento en formato Word y excel,
''con un texto variable al inicio del documento, solicitar ruta y nombre del archivo
''para guardarlo y mostrarlo automaticamente.
Public Class Form1
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles btnGuardar.Click
        Dim ruta, dato As String
        Dim dialogoGuardar As SaveFileDialog = New SaveFileDialog()
        dialogoGuardar.Filter = "Archivos Word (*.docx)|*.docx|Hoja de cálculo de Microsoft Excel (*.xlsx)|*.xlsx"
        If (dialogoGuardar.ShowDialog() <> DialogResult.OK) Then

            Return

        ElseIf (dialogoGuardar.FilterIndex = 1) Then
            ruta = dialogoGuardar.FileName
            Dim WordApp = New Word.Application()
            WordApp.Visible = True
            WordApp.Documents.Add()
            dato = txtDato.Text
            WordApp.Selection.TypeText(dato)
            WordApp.ActiveDocument.SaveAs2(ruta)
            MessageBox.Show("Archivo creado")
        ElseIf (dialogoGuardar.FilterIndex = 2) Then
            Dim aplicacion As Excel.Application
            Dim libros_trabajo As Excel.Workbook
            Dim hoja_trabajo As Excel.Worksheet
            aplicacion = New Excel.Application()
            aplicacion.Visible = True
            libros_trabajo = aplicacion.Workbooks.Add()
            hoja_trabajo = libros_trabajo.Worksheets(1)
            hoja_trabajo.Cells(1, 1) = txtDato.Text
            libros_trabajo.SaveAs(dialogoGuardar.FileName)
            MessageBox.Show("Archivo creado")
        End If

    End Sub
End Class

File: ExcelWordC/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Word= Microsoft.Office.Interop.Word;
using Excel = Microsoft.Office.Interop.Excel;
namespace ExcelWordC
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();

        }

        private void btnGuardar_Click(object sender, EventArgs e)
        {
            string ruta,dato;
            SaveFileDialog dialogoGuardar = new SaveFileDialog();
            dialogoGuardar.Filter = "Archivos Word (*.docx)|*.docx|Hoja de cálculo de Microsoft Excel (*.xlsx)|*.xlsx";
            if(dialogoGuardar.ShowDialog() != DialogResult.OK)
                        {
                return;
            }
            else if(dialogoGuardar.FilterIndex == 1){
                ruta= dialogoGuardar.FileName;
                var WordApp = new Word.Application();
                WordApp.Visible= true;
                WordApp.Documents.Add();
                dato = txtDato.Text;
                WordApp.Selection.TypeText(dato);
                WordApp.ActiveDocument.SaveAs2(ruta);
                MessageBox.Show("Archivo creado");
            }
            else if (dialogoGuardar.FilterIndex == 2)
            {

                Excel.Application aplicacion;
                Excel.Workbook libros_trabajo;
                Excel.Worksheet hoja_trabajo;
                aplicacion = new Excel.Application();
                aplicacion.Visible = true;
                libros_trabajo = aplicacion.Workbooks.Add();
                hoja_trabajo = libros_trabajo.Worksheets.get_Item(1);
                hoja_trabajo.Cells[1, 1] = txtDato.Text;
                libros_trabajo.SaveAs(dialogoGuardar.FileName);
                MessageBox.Show("Archivo creado");
            }

        }

        private void Label2_Click(object sender, EventArgs e)
        {

        }
    }
}
